httputil/cert_ext_tripper.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package httputil
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/asn1"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+type ignoreExtensionsRoundTripper struct {
+	base         *http.Transport
+	extsToIgnore []asn1.ObjectIdentifier
+	logger       hclog.Logger
+}
+
+// NewIgnoreUnhandledExtensionsRoundTripper creates a RoundTripper that may be used in an HTTP client which will
+// ignore the provided extensions if present unhandled on a certificate.  If base is nil, the default RoundTripper is used.
+func NewIgnoreUnhandledExtensionsRoundTripper(logger hclog.Logger, base http.RoundTripper, extsToIgnore []asn1.ObjectIdentifier) http.RoundTripper {
+	if len(extsToIgnore) == 0 {
+		return base
+	}
+	if base == nil {
+		base = http.DefaultTransport
+	}
+
+	tp, ok := base.(*http.Transport)
+	if !ok {
+		// We don't know how to deal with this object, bail
+		return base
+	}
+
+	return &ignoreExtensionsRoundTripper{base: tp, logger: logger, extsToIgnore: extsToIgnore}
+}
+
+func (i *ignoreExtensionsRoundTripper) RoundTrip(request *http.Request) (*http.Response, error) {
+	var domain string
+	if strings.ContainsRune(request.URL.Host, ':') {
```

Comment:
I believe this will fail for IPv6 since it's valid to have an IP address as the host.

You should do a `net.SplitHostPort` and check the returned error, if any, something like:

```
			host, _, err = net.SplitHostPort(ln.Config.Address)
			if err != nil {
				if strings.Contains(err.Error(), "missing port") {
					host = ln.Config.Address
				} else {
					return fmt.Errorf("error splitting host/port: %w", err)
				}
			}
```

Reply:
Good call.